cypress/support/commands.ts

```diff
@@ -58,3 +61,73 @@ Cypress.on('uncaught:exception', (err: Error): boolean | Error => {
 
   return err;
 });
+
+const toJsonDate = (date: Date): string => date.toJSON().split('T')[0];
+
+Cypress.Commands.add(
+  'createDatePeriod',
+  ({
+    name,
+    startDate,
+    endDate,
+    resourceId,
+  }: {
+    name: string;
+    startDate: Date;
+    endDate: Date;
+    resourceId: string;
+  }): Chainable => {
+    return cy
+      .visit('/')
+      .window()
+      .then((win: Window) => {
+        const apiUrl = win.ENV?.API_URL;
+
+        if (!apiUrl) {
+          return cy.task('log', 'API_URL is not set!!!');
+        }
+
+        return cy
+          .exec('node ./scripts/generate-auth-params.js')
+          .then((params) =>
+            cy
+              .request(
+                'GET',
+                `${apiUrl}/v1/resource/${resourceId}?${params.stdout}`
+              )
+              .then((resourceResponse) => {
```

Comment:
I think we could reduce the amount of nested promise chains here by closing the upper then before this line, and then we could continue with a new then that has the same intentation that the previous one has?

Reply:
This is fixed.